### PR TITLE
[Admin] Cover creating and updating Taxon resource

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
|-----------------|-----
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | yes
| Deprecations?   | 
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: features/admin/taxonomy/managing_taxons/editing_taxon_slug.feature [hunk 1]
```diff
@@ -54,97 +57,89 @@ Feature: Editing taxon's slug
...
         And the store has "Pikes" taxonomy
         When I want to modify the "Pikes" taxon
         And I set its parent taxon to "Medieval weapons"
         And I save my changes
         Then the slug of the "Pikes" taxon should still be "pikes"
 
-    @todo @ui @javascript @api
-    Scenario: Prevent from editing the slug while changing the taxon's parent
+#    @api
+    @ui @mink:chromedriver
```
Comment (Wojdylak) [inline features/admin/taxonomy/managing_taxons/editing_taxon_slug.feature]:
```suggestion
    @ui @mink:chromedriver @todo-api
```

File: src/Sylius/Behat/Context/Ui/Admin/ManagingTaxonsContext.php [hunk 2]
```diff
@@ -82,79 +76,84 @@ public function iWantToModifyATaxon(TaxonInterface $taxon)
...
-    public function iChangeItsDescriptionToIn($description, $language)
+    public function iDescribeItAs(string $description, string $localeCode): void
     {
-        $this->updatePage->describeItAs($description, $language);
+        $this->formElement->describeItAs($description, $localeCode);
     }
 
     /**
-     * @When I describe it as :description in :language
+     * @Given /^I set its (parent taxon to "[^"]+")$/
```
Comment (Wojdylak) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingTaxonsContext.php]: `Given` steps before `When` steps

File: src/Sylius/Behat/Element/Admin/Taxon/FormElement.php [hunk 3]
```diff
@@ -32,12 +53,91 @@ public function slugIt(string $slug, string $localeCode): void
...
+    }
+
+    public function getParent(): string
+    {
+        return $this->getElement('parent')->getValue();
+    }
+
+    public function chooseParent(TaxonInterface $taxon): void
+    {
+        $this->autocompleteHelper->removeByName($this->getDriver(), $this->getElement('parent')->getXpath(), '');
```
Comment (Wojdylak) [inline src/Sylius/Behat/Element/Admin/Taxon/FormElement.php]: Why do you need this line?

File: src/Sylius/Bundle/AdminBundle/Resources/config/services.xml [hunk 4]
```diff
@@ -116,5 +116,9 @@
                  form-type="Sylius\Bundle\AttributeBundle\Form\Type\AttributeType\SelectAttributeType"
                  configuration-form-type="Sylius\Bundle\AdminBundle\Form\Type\AttributeType\Configuration\SelectAttributeConfigurationType" />
         </service>
+
+        <service id="sylius_admin.generator.taxon_slug" class="Sylius\Bundle\AdminBundle\Generator\TaxonSlugGenerator">
+            <argument type="service" id="sylius.generator.taxon_slug" />
+        </service>
```
Comment (Wojdylak) [inline src/Sylius/Bundle/AdminBundle/Resources/config/services.xml:122]:
```suggestion
        <service id="sylius_admin.generator.taxon_slug" class="Sylius\Bundle\AdminBundle\Generator\TaxonSlugGenerator">
            <argument type="service" id="sylius.generator.taxon_slug" />
        </service>
        <service id="Sylius\Bundle\AdminBundle\Generator\TaxonSlugGeneratorInterface" alias="sylius_admin.generator.taxon_slug">
```

**Review by Wojdylak — APPROVED** [review]

File: src/Sylius/Bundle/AdminBundle/templates/product/form_theme.html.twig [hunk 5]
```diff
@@ -24,7 +26,7 @@
             data-live-locale-code-param="{{ locale }}"
             {{ sylius_test_html_attribute('generate-product-slug-button', locale) }}
         >
-            ⚡️
+            {{ icon({ icon: 'wand' }) }}
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/AdminBundle/templates/product/form_theme.html.twig:29]: 💯

**Review by NoResponseMate — APPROVED** [review]

**Pull request merged**